lib/ansible/modules/system/beadm.py

```diff
@@ -0,0 +1,376 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Adam Števko <adam.stevko@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: beadm
+short_description: Manage ZFS boot environments on FreeBSD/Solaris/illumos systems.
+description:
+    - Create, delete or activate ZFS boot environments.
+    - Mount and unmount ZFS boot environments.
+version_added: "2.2"
+author: Adam Števko (@xen0l)
+options:
+    name:
+        description:
+            - ZFS boot environment name.
+        alias: [ "be" ]
+        required: True
+    snapshot:
+        description:
+            - If specified, the new boot environment will be cloned from the given
+              snapshot or inactive boot environment.
+        required: false
+        default: false
+    description:
+        description:
+            - Associate a description with a new boot environment. This option is
+              available only on Solarish platforms.
+        required: false
+        default: false
+    options:
+        description:
+            - Create the datasets for new BE with specific ZFS properties. Multiple
+              options can be specified. This option is available only on
+              Solarish platforms.
+        required: false
+        default: false
+    mountpoint:
+        description:
+            - Path where to mount the ZFS boot environment
+        required: false
+        default: false
+    state:
+        description:
+            - Create or delete ZFS boot environment.
+        required: false
+        default: "present"
+        choices: [ "present", "absent", "activated", "mounted", "unmounted" ]
+    force:
+        description:
+            - Specifies if the unmount should be forced.
+        required: false
+        default: false
+        choices: [ "true", "false" ]
+'''
+
+EXAMPLES = '''
+# Create ZFS boot environment
```

Comment:
Please write examples in multi-line YAML with `key: value`
Also  use `name: ...` rather than a comment to describe what the task is doing.